domenetjenester/behandling-revurdering/src/main/java/no/nav/foreldrepenger/behandling/revurdering/ytelse/fp/UttakGrunnlagTjeneste.java

```diff
@@ -106,14 +107,38 @@ public ForeldrepengerGrunnlag grunnlag(BehandlingReferanse ref) {
             .medPleiepengerGrunnlag(pleiepengerGrunnlag(ref).orElse(null))
             .medOppdagetPleiepengerOverlappendeUtbetaling(behandling.harBehandlingÅrsak(BehandlingÅrsakType.RE_VEDTAK_PLEIEPENGER))
             .medUføretrygdGrunnlag(uføretrygdGrunnlag(ref).orElse(null))
-            .medNesteSakGrunnlag(nesteSakGrunnlag(ref).orElse(null));
+            .medNesteSakGrunnlag(nesteSakGrunnlag(ref).orElse(null))
+            .medDødsfall(harDødsfall(behandling, familiehendelser, ref))
+            ;
         if (fagsakRelasjon.isPresent()) {
             var annenpart = annenpart(fagsakRelasjon.get(), behandling);
             grunnlag = grunnlag.medAnnenpart(annenpart.orElse(null));
         }
         return grunnlag;
     }
 
+    private boolean harDødsfall(Behandling behandling, FamilieHendelser familiehendelser, BehandlingReferanse ref) {
+        var barna = familiehendelser.getGjeldendeFamilieHendelse().getBarna();
+
+        var harDødsdatoPåBarn = barna.stream().anyMatch(barn -> barn.getDødsdato().isPresent());
+        var harBehandlingÅrsakMedDød = behandling.getBehandlingÅrsaker()
+            .stream()
+            .anyMatch(årsak -> BehandlingÅrsakType.årsakerRelatertTilDød().contains(årsak.getBehandlingÅrsakType()));
+        return harDødsdatoPåBarn || harBehandlingÅrsakMedDød || erOpplysningerOmDødEndretIPersonopplysninger(ref);
+    }
+
+    private boolean erOpplysningerOmDødEndretIPersonopplysninger(BehandlingReferanse ref) {
+        var behandlingId = ref.behandlingId();
+        var originaltGrunnlag = personopplysningRepository.hentFørsteVersjonAvPersonopplysninger(behandlingId);
+        var nåværendeGrunnlag = personopplysningRepository.hentPersonopplysninger(behandlingId);
+        var poDiff = new PersonopplysningGrunnlagDiff(ref.aktørId(), nåværendeGrunnlag, originaltGrunnlag);
+
+        var barnDødt = poDiff.erBarnDødsdatoEndret();
```

Comment:
Ligger ikke noe tilsvarende i BehandlingÅrsakUtlederPersonopplysning som sørger for at det settes en årsak?
Kan den brukes?

Reply:
spørsmål om vi trenger både i sjekke på årsak, endring OG dødsdato på barn?? så lenge barnet eller forelder er død så skal ha AP i uttak og endringsdato fra startdato

Reply:
Ser ut som BehandlingÅrsakTjeneste ikke setter behandlingårsak, bare lager historikkinnslag ....

Reply:
Tenker at erOpplysningerOmDødEndretIPersonopplysninger er unødvendig. Er forelder/barn i FREG så har den aktuelle registrert dødsdato. Barn uten FNR er en issue ettersom man ikke får overstyre fakta/fødsel - men kan  foreligge i Overstyrt FH. FH er vel i UttakInput mens foreldres døsdato ikke er det. 
Unntaket er nok feilregistrering i FREG med annullert død.